src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -394,7 +394,9 @@ private[collection] final class INode[K, V](bn: MainNode[K, V], g: Gen, equiv: E
   private def clean(nd: INode[K, V], ct: TrieMap[K, V], lev: Int): Unit = {
     val m = nd.GCAS_READ(ct)
     m match {
-      case cn: CNode[K, V] => nd.GCAS(cn, cn.toCompressed(ct, lev, gen), ct)
+      case cn: CNode[K, V] =>
+        nd.GCAS(cn, cn.toCompressed(ct, lev, gen), ct)
+        ()
```

Comment:
see the comment on iterator, this is the same as one-legged if, which is side-effecting. The difference in syntax is that (on the other PR) the empty default case `case _ => ()` is required, where the idea is that surely you didn't intend to lub with unit.